lib/service/service.go

```diff
@@ -4685,6 +4685,12 @@ func (process *TeleportProcess) waitForAppDepend() {
 
 // registerExpectedServices sets up the instance role -> identity event mapping.
 func (process *TeleportProcess) registerExpectedServices(cfg *servicecfg.Config) {
+	// Register additional expected services for this Teleport instance.
+	// Meant for enterprise support.
+	for _, r := range cfg.AdditionalExpectedRoles {
```

Comment:
Also, why do we need to pass this to OSS via AdditionalExpectedRoles, can't enterprise code just use `SetExpectedInstanceRole(...)` on the `process` when initializing appropriate service?

Reply:
It cannot, unfortunately. Basically the inventory status message is created at the very beginning of `NewTeleport`, so the enterprise statuses never actually get the roles from `SetExpectedInstanceRole`.